syne_tune/config_space.py

```diff
@@ -612,38 +612,44 @@ def __init__(self, lower: float, upper: float, size: int,
         self.size = size
         if not log_scale:
             self._lower_internal = lower
-            self._step_internal = (upper - lower) / (size - 1)
+            self._step_internal = (upper - lower) / (size - 1) if size > 1 else 0
         else:
             self._lower_internal = np.log(lower)
             upper_internal = np.log(upper)
-            self._step_internal = \
-                (upper_internal - self._lower_internal) / (size - 1)
+            self._step_internal = (upper_internal - self._lower_internal) / (size - 1) if size > 1 else 0
+        self._values = [self._map_from_int(x) for x in range(self.size)]
```

Comment:
Could you make this a property, say self.values, and then use it in the function below?

Reply:
I chose this way as it allows to cache the values when sampling.

Reply:
Yes, I like it, but maybe we should have a property, otherwise you access something internal below

Reply:
I am not sure I follow, `_values` is accessed only by the class right?

Reply:
Oh I see, you mean in the restrict domain method, I will add a property as well.